tests/cmd/auth/test_logout.py

```diff
@@ -0,0 +1,206 @@
+
+        prepare_config()
+        exit_code, output = self.run_cmd(cli_fs_runner)
+
+        post_mock.assert_called_once()
+        config = Config()
+        assert config.auth_config.get_instance(DEFAULT_INSTANCE_URL).account is not None
+
+        assert exit_code == 1, output
+        assert output == (
+            "Error: Logout failed due to an error when revoking the token, "
+            "you can skip revocation with --no-revoke to bypass\n"
+        )
+
+    def test_logout_server_error(self, monkeypatch, cli_fs_runner):
+        """
+        GIVEN a saved instance configuration
+        WHEN running the logout command (with implied token revokation)
+        AND the revoke request gets a server error response
+        THEN the config remains unchanged
+        AND the command exits with an explanatory message
+        """
+        post_mock = Mock(return_value=Mock(status_code=500, ok=False))
+        monkeypatch.setattr("ggshield.core.client.GGClient.post", post_mock)
+
+        prepare_config()
+        exit_code, output = self.run_cmd(cli_fs_runner)
+
+        post_mock.assert_called_once()
+        config = Config()
+        assert config.auth_config.get_instance(DEFAULT_INSTANCE_URL).account is not None
+
+        assert exit_code == 1, output
+        assert output == (
+            "Error: Logout failed due to an error when revoking the token, "
+            "you can skip revocation with --no-revoke to bypass\n"
+        )
+
+    def test_logout_all(self, monkeypatch, cli_fs_runner):
+        """
+        GIVEN several saved instances
+        WHEN running the logout command with --all parameter
+        THEN all tokens are revoked
+        AND all account configs are deleted
+        """
+        post_mock = Mock(return_value=Mock(status_code=204, ok=True))
+        monkeypatch.setattr("ggshield.core.client.GGClient.post", post_mock)
+
+        for instance_url in [
+            None,
+            "https://some-gg-instance.com",
+            "https://some-other-gg-instance.com",
+        ]:
+            prepare_config(instance_url)
+
+        exit_code, output = self.run_cmd(cli_fs_runner, all_tokens=True)
+        assert len(post_mock.call_args_list) == 3
+
+        for instance in Config().auth_config.instances:
+            assert instance.account is None, output
+
+        assert exit_code == 0, output
+
+    @staticmethod
+    def run_cmd(
+        cli_fs_runner,
+        instance: Optional[str] = None,
+        no_revoke: bool = False,
+        all_tokens: bool = False,
+    ) -> None:
+        cmd = ["auth", "logout"]
+        if instance is not None:
+            cmd.append("--instance=" + instance)
+        if no_revoke:
+            cmd.append("--no-revoke")
+        if all_tokens:
+            cmd.append("--all")
+        result = cli_fs_runner.invoke(cli, cmd, color=False)
+        return result.exit_code, result.output
+
+
+# TODO: mutualize this with test_login.py
```

Comment:
I think you forgot to remove this function and uses the one you added to utils.py :)